### PR TITLE
AnupS_Assignment_20_Done_50mins

### PR DESCRIPTION
**Pull request by AnupSJun20**
AnupS_Assignment_20_Done_50mins_Please review.
Program1
Program2
Program3
Program4

### CODE REVIEW

File: src/anupSJun20/ArmstrongNumber.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+hint:- (1*1*1)+(5*5*5)+(3*3*3)
+1+125+27
+153*/
+
+public class ArmstrongNumber {
+	static void findANumberArmstrong(int num) {
+		int input = num;
+		int newNum = 0;
+		while(num>0) {
+			newNum = newNum+ (num%10)*(num%10)*(num%10);
```
Comment (AditiGJUN20) [inline src/anupSJun20/ArmstrongNumber.java]: **Suggestion:**
You can take a look at power function - Math.pow(x,y)
which will help you achieve same result for multiplication.
i.e. Math.pow((number % 10), 3))

File: src/anupSJun20/MissingNumbrinIntegerArray.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+		for(int index = 0;index<num.length;index++) {
+			sumofArray = sumofArray+num[index];
+		}
+		return overallSum - sumofArray;
+		
+		
+	}
+	public static void main(String[] args) {
+		int[] num = {1,2,3,4,5,7,10,9,8};
+		System.out.println("The missing number is : " + findTheMissingNumber(num,1,10));
```
Comment (AditiGJUN20) [inline src/anupSJun20/MissingNumbrinIntegerArray.java:18]: @AnupSJun20 
Have you tried to make code generic by passing extra arguments in method?
If yes then - 
for ->int[] num = {11,15,12,13} when i call findTheMissingNumber(num,11,15) it's giving me incorrect result.

File: src/anupSJun20/ReverseNumber.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+/* Program 3: WAP to reverse number (Consider Input as Integer, not as String).
+Example:-
+Input:- 12345
+Output:- 54321*/
+
+public class ReverseNumber {
+	static void findReverseNumber(int num) {
+		int reverseNum = 0;
+		while(num > 0) {
+			reverseNum = reverseNum*10;
```
Comment (AditiGJUN20) [inline src/anupSJun20/ReverseNumber.java]: - You can reduce the statement by following syntax:
reverseNum=reverseNum*10+(inputNum%10);

File: src/anupSJun20/MissingNumbrinIntegerArray.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+public class MissingNumbrinIntegerArray {
+	static int findTheMissingNumber(int[] num,int startNum,int endNum) {
+		int overallSum = (endNum*(endNum+1))/2;
+		int sumofArray = 0;
+		for(int index = 0;index<num.length;index++) {
+			sumofArray = sumofArray+num[index];
+		}
+		return overallSum - sumofArray;
+
```
Comment (AditiGJUN20) [inline src/anupSJun20/MissingNumbrinIntegerArray.java]: remove extra lines

**Review by AditiGJUN20 — CHANGES_REQUESTED** [review]
- I've added separate comment where changes are required.
- Overall for all program remove extra blank lines.
i.e. for MissingNumbrinIntegerArray.java i've added comment.

File: src/anupSJun20/ArmstrongNumber.java (hunk 1 above)
Comment (AnupSJun20) [reply src/anupSJun20/ArmstrongNumber.java]: Thanks Aditi. Changes incorporated. Didn't think of this function.

File: src/anupSJun20/MissingNumbrinIntegerArray.java (hunk 2 above)
Comment (AnupSJun20) [reply src/anupSJun20/MissingNumbrinIntegerArray.java:18]: Thanks Aditi for this comment. I believe Maulik said that try and take a hard coded range 1- 10 as of now and he will explain in the class for user defined values and multiple missing numbers. So once Maulik will explain today, will create a new program. 
Yes the code gave you incorrect result as it was created for a hard code value to start from 1.

File: src/anupSJun20/ReverseNumber.java (hunk 3 above)
Comment (AnupSJun20) [reply src/anupSJun20/ReverseNumber.java]: Thanks for the comment. Incorporated the changes and pushed.

File: src/anupSJun20/MissingNumbrinIntegerArray.java (hunk 4 above)
Comment (AnupSJun20) [reply src/anupSJun20/MissingNumbrinIntegerArray.java]: Removed now.

File: src/anupSJun20/ArmstrongNumber.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+hint:- (1*1*1)+(5*5*5)+(3*3*3)
+1+125+27
+153*/
+
+public class ArmstrongNumber {
+	static void findANumberArmstrong(int num) {
+		int input = num;
+		double newNum = 0;
+		while(num>0) {
+			newNum = newNum + Math.pow((num%10), 3);
```
Comment (AditiGJUN20) [inline src/anupSJun20/ArmstrongNumber.java:15]: There is no need to change data type of newNum from int to double due to power function.
you can use typecasting here.
newNum = (int) (newNum + (Math.pow((number % 10), 3)));

File: src/anupSJun20/MissingNumbrinIntegerArray.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
+package anupSJun20;
+/*Program 1: Find the missing number in integer array of 1 to 10?
+Hint: You have given an integer array which contains numbers from 1 to 10 but one number is missing, 
+you need to write a Java program to find that missing number in an array.*/
+
+public class MissingNumbrinIntegerArray {
+	static int findTheMissingNumber(int[] num,int startNum,int endNum) {
```
Comment (AditiGJUN20) [inline src/anupSJun20/MissingNumbrinIntegerArray.java:7]: **// Suggestion -**
Here, in entire method you are not using startNum integer(neither in calculation or in any SOP statements). So there is no need to pass it in method argument and you can reduce the argument as **static int findTheMissingNumber(int[] num, int endNum)**

Another way is, just pass input array in method argument- **static int findTheMissingNumber(int[] num)** and you can use **int endNum = array.length+1**

File: src/anupSJun20/MissingNumbrinIntegerArray.java (hunk 2 above)
Comment (AditiGJUN20) [reply src/anupSJun20/MissingNumbrinIntegerArray.java:18]: Yes Anup, this will work for hardcoded value 1-10.

**Review by AditiGJUN20 — APPROVED** [review]
Added suggestion for future code reference.
Rest Code looks good. 
Approved for further review and merge.

**Pull request merged**